### ISSUE TITLE
Git Merge Shortcuts Disappear in Some Conditions

### ISSUE THREAD

**Issue by ZhangPuXi** [microsoft/vscode#82946]
- VSCode Version: 1.39.2
- OS Version: Mac OSX

When I merge my code, there's a conflict in a js file.

```const path = require('path');
const os = require('os');

//动态获取本地IP
const getLocalIpAddress = () => {
  let result;

  const network = os.networkInterfaces();

  for (let key in network) {
    const networkUnit = network[key];

    networkUnit.map((item) => {
      if (item.family === 'IPv4' && !item.internal) {
        result = item.address;
      }
    });
  }

  return result;
};

module.exports = {
<<<<<<< HEAD
  entry: '/Invoice/app.js',
=======
<<<<<<< HEAD
  entry: '/explainPages/app.js',
>>>>>>> cd2f036b783fb28030c83110d5c46ab791d66421
  //无需手动配置IP
  localhost: getLocalIpAddress(),
  // localhost: '192.168.1.112',

<<<<<<< HEAD
  buildRootPath: '/gtgjwap/Vue/Invoice/',
  //为了实现自适应用的模板
  template: path.resolve(__dirname, '../index.html')
=======
  buildRootPath: '/gtgjwap/Vue/explainPages/',
  //为了实现自适应用的模板
  template: path.resolve(__dirname, '../index-adaptive.html')
=======
  entry: '/welfareCenter/app.js',
  //无需手动配置IP
  localhost: getLocalIpAddress(),
  //localhost: 'localhost',
  wtestBase: 'huangl',
  buildRootPath: '/gtgjwap/Vue/welfareCenter/',
  //为了实现自适应用的模板
  template: path.resolve(__dirname, '../src/welfareCenter/index.html')
>>>>>>> c7a461f1cabd09d662e8f9cec622a99872db66bb
>>>>>>> cd2f036b783fb28030c83110d5c46ab791d66421
}
```

And the Accept Incoming Changes button is disappear.

When I start manually fix the conflict, just delete the first <<<<<<< HEAD line, the shortcuts is appear. I'm not sure whether it is a bug. Just report for your information.

**Comment by chrmarti** [comment]
The merge conflicts appear to overlap. Can you provide step-by-step instructions on how to arrive at the above content?

**Comment by ZhangPuXi** [comment]
1. Copy the above code to a new js file. And all you can see is just raw code.

![image](https://user-images.githubusercontent.com/15119639/67194063-fb177b00-f428-11e9-84b0-ae267a6cdd23.png)

2. Remove the first appearance of <<<<<<< HEAD, and you can see the convenient buttons now.

![image](https://user-images.githubusercontent.com/15119639/67194126-18e4e000-f429-11e9-8733-5dc457df6479.png)

I guess this is caused by embed of the '<<<<<<<' strings. Thanks for your time.

**Comment by chrmarti** [comment]
@ZhangPuXi I don't understand what Git (or other SCM) operations resulted in the file content you have. It looks like the overlapping merge conflicts are not valid merge conflict syntax and for that reason not supported.

Starting from a file without merge conflicts, how did you arrive at the file content you have?

**Comment by ZhangPuXi** [comment]
@chrmarti I just merge my dev branch into the master branch and that results in the conflicts. This is the first time I've met this situation. And one of my workmates also met this.

**Comment by chrmarti** [comment]
Maybe one of the commits had one of the merge conflict texts as its content and when merging that a new merge conflict occurred. The merge conflict syntax does not support nesting / interleaving of conflicts, so I will close this issue. Thanks.

**Comment by sarpik** [comment]
Hello, I'd like to add to this - the shortcuts disappear for me too! This happens always since I've updated vscode from 1.38 to 1.39.2.

I opened the same merge conflict with vscode 1.38 / vscode-insiders 1.39.0 and it works absolutely fine.

Can we get this re-opened?

/cc @chrmarti 

Let's create a sample conflict:

```sh
mkdir repo
cd repo
git init
touch file
echo "1" > file
git add file
git commit -m "master: init & write '1' to file"
git checkout -b other-branch
echo "2" > file
git add file
git commit -m "other-branch: change '1' to '2' in file"
git checkout master
echo "3" > file
git add file
git commit -m "master again: change '1' to '3' in file"
git checkout other-branch
git rebase -i master
# <confirm>
```

```console
Auto-merging file
CONFLICT (content): Merge conflict in file
error: could not apply 0d65af8... other branch: change '1' to '2' in file
Resolve all conflicts manually, mark them as resolved with
"git add/rm <conflicted_files>", then run "git rebase --continue".
You can instead skip this commit: run "git rebase --skip".
To abort and get back to the state before "git rebase", run "git rebase --abort".
Could not apply 0d65af8... other branch: change '1' to '2' in file
```

```sh
cat file
```

```console
<<<<<<< HEAD
3
=======
2
>>>>>>> 0d65af8... other branch: change '1' to '2' in file
```

```sh
code .
```

![vscode-normal-nope](https://user-images.githubusercontent.com/29430509/67404265-c982e900-f5bb-11e9-9fa8-46e12ea32f7c.png)


As you can see - the `resolve merge conflict` buttons are nowhere to be found. There is no way in hell that vscode cannot support this.

Now, if I open up an older version of vscode (`1.39.0-insider` instead of `1.39.2
`), it all works fine:

![vscode-insiders-works-thought](https://user-images.githubusercontent.com/29430509/67404280-ce479d00-f5bb-11e9-87be-afac7747701d.png)

it will probably work with vscode `1.38.x`.

---

My setup:

```console
$ uname -a
Linux arch-usb 5.3.7-arch1-1-ARCH #1 SMP PREEMPT Fri Oct 18 00:17:03 UTC 2019 x86_64 GNU/Linux

$ code -v
1.39.2
6ab598523be7a800d7f3eb4d92d7ab9a66069390
x64
```

**Comment by sarpik** [comment]
Fixed in #83221!
